src/routes/profile.js

```diff
@@ -53,132 +55,113 @@ router.get('/mod/customavatar', isModMiddleware, (req, res) => {
-    if (err) {
-      console.log(err);
-    } else if (foundReq) {
-      foundReq.processed = true;
-      foundReq.modComment = req.body.modcomment;
-      foundReq.approved = req.body.decision;
-      foundReq.modWhoProcessed = req.user.username;
-
-      if (req.body.decision === true || req.body.decision === 'true') {
-        const updateLink = async (link) => {
-          const index = link.indexOf('pending_avatars');
-          const endpoint = link.substring(0, index);
-          const key = link.substring(index);
-
-          const updatedKey = await s3.approveAvatarRefactorFilePath(key);
-
-          return `${endpoint}${updatedKey}`;
-        };
-
-        foundReq.resLink = await updateLink(foundReq.resLink);
-        foundReq.spyLink = await updateLink(foundReq.spyLink);
-
-        foundReq.markModified('resLink');
-        foundReq.markModified('spyLink');
-
-        await foundReq.save();
-
-        User.findOne({ usernameLower: foundReq.forUsername.toLowerCase() })
-          .populate('notifications')
-          .exec((err, foundUser) => {
-            if (err) {
-              console.log(err);
-            } else {
-              foundUser.avatarImgRes = foundReq.resLink;
-              foundUser.avatarImgSpy = foundReq.spyLink;
-
-              foundUser.save();
-
-              let str = `Your avatar request was approved by ${foundReq.modWhoProcessed}!`;
-              if (foundReq.modComment) {
-                str += ` Their comment was: ${foundReq.modComment}`;
-              }
-
-              createNotification(foundUser._id, str, '#', req.user.username);
-            }
-          });
-      } else if (req.body.decision === false || req.body.decision === 'false') {
-        const pattern = /pending_avatars\/.*$/;
-
-        if (
-          !foundReq.resLink.match(pattern) ||
-          !foundReq.resLink.match(pattern)
-        ) {
-          throw new Error(
-            `Invalid link provided: resLink: ${foundReq.resLink} spyLink: ${foundReq.spyLink}`,
-          );
-        }
-
-        await s3.rejectAvatarRequest(foundReq.resLink.match(pattern)[0]);
-        await s3.rejectAvatarRequest(foundReq.spyLink.match(pattern)[0]);
+router.post(
+  '/mod/ajax/processavatarrequest',
+  isModMiddleware,
+  async (req, res) => {
+    const avatarReq = await avatarRequest.findById(req.body.avatarreqid);
+    const userRequestingAvatar = await User.findOne({
+      usernameLower: avatarReq.forUsername.toLowerCase(),
+    });
 
-        foundReq.resLink = null;
-        foundReq.spyLink = null;
+    const modWhoProcessed = req.user;
 
-        User.findOne({ usernameLower: foundReq.forUsername.toLowerCase() })
-          .populate('notifications')
-          .exec((err, foundUser) => {
-            if (err) {
-              console.log(err);
-            } else {
-              let str = `Your avatar request was rejected by ${foundReq.modWhoProcessed}.`;
+    if (req.body.decision === true || req.body.decision === 'true') {
```

Comment:
Can you try to remove the `|| req.body.decision === 'true'` and see if it still works? 
Can we check whether req.body.decision comes in as a bool pls?

Reply:
If removing it works, let's also remove the else if for the false case and throw it into else.

Reply:
it comes in as a string

Reply:
TODO: Cast it to a bool, throw an error where decision is not recognisable

Reply:
Done